via/views/blocker.py

```diff
@@ -35,3 +37,33 @@ def view_wrapper(context, request):
         return view(context, request)
 
     return view_wrapper
+
+
+def secure_url_token(view, url_param="sec"):
+    def view_wrapper(context, request):
+        if not request.registry.settings["signed_urls_enabled"]:
+            return view(context, request)
+
+        secure_token = ViaSecureURLToken(request.registry.settings["via_secret"])
+
+        if url_param not in request.params:
+            logging.info(
+                "ViaSecureURLToken parameter (%s) not present in the URL", url_param
+            )
+            return HTTPUnauthorized()
+
+        # Remove the signing token from the URL to verify
+        url_token = request.params[url_param]
+        signed_url = urlparse(request.url)
+        options = dict(request.params)
+        del options[url_param]
+        signed_url = signed_url._replace(query=urlencode(options)).geturl()
+
+        try:
+            secure_token.verify(request.params["sec"], signed_url)
+        except (MissingToken, InvalidToken):
+            return HTTPUnauthorized()
```

Comment:
```suggestion
def secure_url_token(view, url_param="via.sec"):
    def view_wrapper(context, request):
        if not request.registry.settings["signed_urls_enabled"]:
            return view(context, request)

        secure_token = ViaSecureURLToken(request.registry.settings["via_secret"])
        
        try:
            secure_token.verify(request.params.get(url_param), request.url)
        except TokenException:
            return HTTPUnauthorized()
            
        return None
```

Quite a few things here:

* `ViaSecureURLToken` already raises `MissingToken` if passed None, as we don't need to check for that
* We don't need to clean the URL as:
    *  ViaSecureURLToken already does
    * There's nothing in the URL in the first place to clean. The URL param is totally separate from the token
* We use `via.*` for via configuration parameters. It seems like it might simplify things to do this in general
* We have a unified base class for the exceptions now!

Reply:
I'm not sure about the names of the decorators here. "secure_url_token" reads as a noun to me rather than an action. "has_secure_url_token" or "with_url_signature" or something?